src/layouts/ReviewRequest/components/Comments/CommentsDrawer.tsx

```diff
@@ -0,0 +1,174 @@
+  const {
+    isOpen: isCommentsOpen,
+    onOpen: onCommentsOpen,
+    onClose: onCommentsClose,
+  } = useDisclosure()
+
+  const {
+    data: commentsData,
+    isLoading: isCommentsLoading,
+    refetch: refetchRecentNotificationData,
+  } = useGetComments({
+    siteName,
+    requestId,
+  })
+
+  const { mutateAsync: updateReadComments } = useUpdateReadComments()
+  const {
+    mutateAsync: updateNotifications,
+    error: updateNotificationsError,
+  } = useUpdateComments()
+
+  const handleUpdateNotifications = async ({ comment }: CommentFormProps) => {
+    await updateNotifications({ siteName, requestId, message: comment })
+  }
+
+  useEffect(() => {
+    refetchRecentNotificationData()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return (
+    <>
+      <IconButton
+        backgroundColor="background.action.defaultInverse"
+        onClick={() => {
+          onCommentsOpen()
+          // updateReadComments({ siteName, requestId })
+        }}
+        aria-label="Open comments"
+        icon={<BiCommentDetail />}
+        variant="clear"
+        boxShadow="0 0 0 2px var(--chakra-colors-gray-100)"
+      />
+      <Drawer
+        isOpen={isCommentsOpen}
+        onClose={onCommentsClose}
+        placement="right"
+        size="sm"
+      >
+        <DrawerOverlay />
+        <DrawerContent px="1rem">
+          <DrawerCloseButton
+            variant="clear"
+            colorScheme="secondary"
+            top="1.25rem"
+          />
+          <DrawerHeader fontSize="1.5rem" color="text.title.alt">
+            Comments
+          </DrawerHeader>
+          <DrawerBody
+            fontSize="0.875rem"
+            whiteSpace="pre-line"
+            color="text.body"
+          >
+            <Stack spacing="2rem" height="100%">
+              <Text color="text.helper">
+                Comments apply to all items in this Review request
+              </Text>
+              <Skeleton isLoaded={!isCommentsLoading} flex={1}>
+                {commentsData && commentsData.length > 0 ? (
+                  commentsData.map((comment) => (
+                    <CommentItem
+                      commenterName={comment.email}
+                      commentTime={comment.createdAt}
+                      isNew={comment.isNew}
+                    >
+                      {comment.message}
+                    </CommentItem>
+                  ))
+                ) : (
+                  <Flex
+                    flex={1}
+                    h="100%"
+                    justifyContent="center"
+                    alignItems="center"
+                  >
+                    <EmptyChatImage />
+                  </Flex>
+                )}
+              </Skeleton>
+              <Divider borderColor="border.divider.alt" />
+              <SendCommentForm
+                onSubmit={handleUpdateNotifications}
+                errorMessage={getAxiosErrorMessage(updateNotificationsError)}
```

Comment:
could i ask why we're putting this (the hook call) in the parent rather than just in `SendCommentForm` as idt it's used in the parent

Reply:
yeah that makes sense, https://github.com/isomerpages/isomercms-frontend/pull/1102/commits/adb6ad89720199f837019680ae87bdd6d15d6324